client/extensions/woocommerce/state/data-layer/request/index.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * Internal dependencies
+ */
+import request from 'woocommerce/state/sites/request';
+import { setError } from 'woocommerce/state/sites/status/wc-api/actions';
+
+import {
+	WOOCOMMERCE_REQUEST,
+	WOOCOMMERCE_REQUEST_SUCCESS,
+	WOOCOMMERCE_REQUEST_FAILURE,
+} from 'woocommerce/state/action-types';
+
+export function handleRequest( { dispatch }, action ) {
+	const { method, siteId, path, body } = action;
+
+	return request( siteId )[ method ]( path, body )
+		.then( data => {
+			dispatch( {
+				type: WOOCOMMERCE_REQUEST_SUCCESS,
+				action,
+				data,
+			} );
+		} )
+		.catch( error => {
+			// TODO: Maybe phase out usage of this in favor of the failure action?
+			dispatch( setError( siteId, action, { message: error.toString() } ) );
+			dispatch( {
+				type: WOOCOMMERCE_REQUEST_FAILURE,
+				action,
+				error,
+			} );
+		} );
+}
+
+export default {
```

Comment:
Yeah, after coding up all this data-layer stuff, I find myself wanting to rename all of these files `handler.js` and place them alongside their `actions.js` and `reducer.js` counterparts. I hate to make such big changes to all of these PRs right now though, so I'm planning on a refactoring PR after these are all in and our schedule is maybe not so pressing.